src/types/permissions.d.ts

```diff
@@ -30,3 +30,12 @@ export interface IdpGroup {
   name: string;
   groups: string[]; // these should be names of lxd groups
 }
+
+export interface LxdFineGrainedTlsIdentity {
+  client_name: string;
+  addresses: string[];
+  expires_at: string;
+  fingerprint: string;
+  type: "oidc" | "Client certificate (pending)" | "Client certificate";
```

Comment:
I think `type` can't be `oidc` in this case since this data gets returned from the `.../tls` endpoint. On second thought, maybe we can change the type name to `LxdTlsIdentity` instead, it's less verbose

Reply:
Surely the type name should remain LxdFineGrainedTlsIdentity as "LxdTlsIdentity is too broad. It does not consider Unrestricted TLS identities.

Reply:
I've done some digging look at the API [docs](https://documentation.ubuntu.com/lxd/en/latest/api/#/), please refer to it for the following observation:

1. The naming for the type is wrong, since `LxdFineGrainedTlsIdentity` is a very specific response data struct only returned by making a request to `POST /1.0/auth/identities/tls`. The response does not represent an "official" identity data structure in LXD, and it includes details like `addresses`, `expires_at`, `fingerprint` and `secret` etc because those are required to authenticate a TLS identity later. Instead, we should think of a specific type name for the response data from this endpoint, perhaps what you had earlier was a closer fit e.g. `TlsIdentityTokenDetail`.

2. The official identity data structure from the LXD backend can be retrieved from `GET /1.0/auth/identities?recursion=1`, which is already represented by the existing `LxdIdentity` type. Can you please augment the existing type by looking at the response data from the documentation, add what is missing and also make the `type` field more strict (you will need to look at the network request for this since the documentation isn't all comprehensive).

3. After looking at the documentation for the `POST /1.0/auth/identities/tls` endpoint, I see we can set a `certificate` field in the request payload, instead of `token`. Please would you investigate what happens if we set `certificate` to `true`. I suspect we can potentially skip the create certificates in the UI step with this (for future reference).